### PR TITLE
perf(compaction): adjust restart points in block builder

### PR DESCRIPTION
**Pull request by soundOfDestiny**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

State table of GlobalSimpleAgg will be only visited by iter_all, thus we can reduce restart points in block builder of them because we will never seek from middle part of these tables.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Review by Little-Wallace — COMMENTED** [review]
Good Idea. Is there any benchmark ?

File: src/storage/src/hummock/sstable/builder.rs [hunk 1]
```diff
@@ -223,8 +227,11 @@ impl<W: SstableWriter> SstableBuilder<W> {
         self.total_key_count += 1;
         self.last_table_stats.total_key_count += 1;
 
-        self.block_builder
-            .add(self.raw_key.as_ref(), self.raw_value.as_ref());
+        self.block_builder.add(
+            self.raw_key.as_ref(),
+            self.raw_value.as_ref(),
+            self.not_only_visited_by_vnode_fetch,
```
Comment (Little-Wallace) [inline src/storage/src/hummock/sstable/builder.rs:233]: I prefer pass it when we create `block_builder`.

Comment (soundOfDestiny) [reply src/storage/src/hummock/sstable/builder.rs]: > I prefer pass it when we create `block_builder`.

fixed

**Comment by soundOfDestiny** [comment]
> Good Idea. Is there any benchmark ?

seems hard because state of `max/min` aggregator group in `GlobalSimpleAgg` is seldom reloaded

File: src/storage/hummock_sdk/src/compaction_group/mod.rs [hunk 2]
```diff
@@ -32,10 +32,12 @@ pub enum StaticCompactionGroupId {
...
     /// All states goes to here by default.
     StateDefault = 2,
+    /// States which we visit them by fetching all KVs of certain vnode(s) goes to here.
+    StateVnodeVisit = 3,
     /// All MVs goes to here.
-    MaterializedView = 3,
+    MaterializedView = 4,
     /// Larger than any `StaticCompactionGroupId`.
-    End = 4,
+    End = 5,
```
Comment (zwang28) [inline src/storage/hummock_sdk/src/compaction_group/mod.rs:40]: Shall we make `End` larger to reserve more static ids for the future?

**Pull request closed without merging**